lib/helpers/upload.js

```diff
@@ -18,14 +33,58 @@ module.exports = function upload(file, anyfetchClient, accessToken, cb) {
         .expect(200)
         .end(cb);
     },
-    function processFolderContent(content, cb) {
-      console.log(content);
-      cb(null);
+    function handleMetada(metadata, cb) {
+      metadata = metadata.body;
+      // Let's build the file address for the user
+      document.identifier = 'https://app.box.com/files/0/f/' + metadata.parent.id + '/1/f_' + metadata.id;
+      document.actions.show = document.identifier;
+      // do not download the filenamee when it is scheduled for deletion or if it is bigger than 50Mb
+      if(!metadata.trashed_at && metadata.size < 52428800) {
+        fileConfig.filename = metadata.name;
+        fileConfig.knownLength = metadata.size;
+        document.creation_date = metadata.created_at;
+        document.modification_date = metadata.modified_at;
+        document.metadata = {
+          title: metadata.name
+        };
+        cb(null);
+      }
+      else {
+        cb('Error: File will be soon deleted');
+      }
+    },
+    function getFileUrl(cb) {
+      request(apiUrl)
+        .get(url + '/content/')
+        .set('Content-Type', 'application/x-www-form-urlencoded')
+        .set('Authorization', 'Bearer ' + accessToken)
+        .expect(200)
+        .end(function(err, res) {
```

Comment:
I actually cant in this case.
The response can either be a 302 or 202 as documented in the box api documentation.